pages/common/routers/form.js

```diff
@@ -67,16 +66,18 @@ module.exports = ({
   };
 
   const _getValues = (req, res, next) => {
-    req.form.values = {
-      ...flattedNested(req.model, req.form.schema),
-      ...req.session.form[req.model.id].values
-    };
+    req.form.values = Object.assign(
+      flattenNested(req.model, req.form.schema),
+      req.session.form[req.model.id].values
+    );
     return getValues(req, res, next);
   };
 
   const _process = (req, res, next) => {
-    req.form.values = mapValues(req.form.schema, ({ format }, key) => {
-      return format ? format(req.body[key]) : req.body[key] || null;
+    req.form.values = mapValues(req.body, (value, key) => {
+      return req.form.schema[key] && req.form.schema[key].format
+        ? req.form.schema[key].format(value)
+        : value || null;
```

Comment:
I'm not sure why the responsibility is reversed here to automatically persist _any_ keys that exist on the body. I'd keep it so it's whitelisted to only the keys that are on the schema, unless there's a compelling reason to do otherwise?

Also, that ternary is getting pretty beefy - I think I'd be happier splitting it out a bit into multiple statements.

Reply:
there _was_ a reason for this. I can't remember what it was though so I'll revert it and see if anything bad happens

Reply:
Ah, so switching it breaks restrictions, as everything in the schema is set to req.form.values, even if it isn't included in the form - a la notes. By looping over just req.body we only set the fields sent in the form

Reply:
I guess it would be better to loop over the schema, and ignore any that aren't included in the form

Reply:
I'm not sure I understand the difference between "the schema" and "the form". Isn't "the schema" the thing we use to configure the form?

Reply:
kind of, notes isn't in the form but it is in the schema, as it is in other places, like the confirm page. I have updated now to only update values received in req.body, and appear in schema